interp/src/debugger/commands/path_parser.rs

```diff
@@ -0,0 +1,115 @@
+use super::{core::ParseNodes, ParsePath};
+
+use pest_consume::{match_nodes, Error, Parser};
+
+type ParseResult<T> = std::result::Result<T, Error<Rule>>;
+type Node<'i> = pest_consume::Node<'i, Rule, ()>;
+
+// include the grammar file so that Cargo knows to rebuild this file on grammar changes
+const _GRAMMAR: &str = include_str!("path_parser.pest");
+
+#[derive(Parser)]
+#[grammar = "debugger/commands/path_parser.pest"]
+
+pub struct PathParser;
+
+#[pest_consume::parser]
+impl PathParser {
+    fn EOI(_input: Node) -> ParseResult<()> {
+        Ok(())
+    }
+
+    fn root(_input: Node) -> ParseResult<()> {
+        Ok(())
+    }
+
+    fn body(_input: Node) -> ParseResult<()> {
+        Ok(())
+    }
+
+    fn seperator(_input: Node) -> ParseResult<()> {
+        Ok(())
+    }
+
+    fn num(input: Node) -> ParseResult<u32> {
+        input
+            .as_str()
+            .parse::<u32>()
+            .map_err(|_| input.error("Expected non-negative number"))
+    }
+
+    fn branch(input: Node) -> ParseResult<bool> {
+        let b = input.as_str();
+        let result = b != "f";
+        Ok(result)
+    }
+
+    fn clause(input: Node) -> ParseResult<ParseNodes> {
+        Ok(match_nodes!(input.into_children();
+            [seperator(_), num(n)] => ParseNodes::Offset(n),
+            [seperator(_), body(_)] => ParseNodes::Body,
+            [seperator(_), branch(b)] => ParseNodes::If(b)
+        ))
+    }
+
+    fn path(input: Node) -> ParseResult<ParsePath> {
+        Ok(match_nodes!(input.into_children();
+            [root(_), clause(c).., EOI(_)] => ParsePath::from_iter(c),
+        ))
+    }
+}
+
+// Parse the path
+pub fn parse_path(input_str: &str) -> Result<ParsePath, Error<Rule>> {
+    let entries = PathParser::parse(Rule::path, input_str)?;
+    let entry = entries.single()?;
+
+    PathParser::path(entry)
+}
+
```

Comment:
add `#[allow(dead_code)]` here for the moment since this PR doesn't use the
function. This will quiet clippy down.

Reply:
To fix the size issue change the return type to `Box<Error<Rule>>` and add
`.map_err(Box::new)` to the final line